src/Pact/Types/Type.hs

```diff
@@ -348,6 +393,7 @@ canUnifyWith (TyPrim (TyGuard a)) (TyPrim (TyGuard b)) = case (a,b) of
   (Nothing,Just _) -> True
   (Just _,Nothing) -> True
   _ -> a == b
+canUnifyWith (TyModSpec a) (TyModSpec b) = all (`elem` b) a
```

Comment:
Doesn't this need to be symmetric?

```haskell
all (`elem` b) a && all (`elem` a) b
```

Reply:
See the unit tests -- I would assume it's ok/desired to be able to provide an _over-specified_ modref (ie, spec calls for `ifaceA`, value implements `ifaceA` + `ifaceB`)

Reply:
That makes sense, but I'm wondering whether typechecking is the right phase to check that `A ⊆ A + B`. At this point, wouldn't we have resolved the schematype of the signatures and want equality? Or will this happen after we establish that fact?

Reply:
This is runtime typechecking only, so the example would be e.g. in `prepareUserAppArgs` in Eval https://github.com/kadena-io/pact/blob/2ca4cae3eb8a0ef2783aa9457388c31b44a0dfbc/src/Pact/Eval.hs#L719

A typical use case would be user transaction code (as opposed to loaded module code) calling a function `foo` with an arg `module{ifaceA}`, but the module that has been passed implements `ifaceA` and `ifaceB`. Obviously, the module is compliant.

At the `foo` callsite, the module reference will be typed `module{ifaceA,ifaceB}`, and the runtime will call `canUnifyWith` to see if it unifies with the arg, of type `module{ifaceA}`.

Reply:
Okay, agreed then this is fine :)